### DISCUSSION TITLE
One-to-one relationship should be supported

### DISCUSSION THREAD

**Discussion by birolozturkk** [j256/ormlite-core#268] [Ideas]
```java
@AllArgsConstructor
@NoArgsConstructor
@Data
@DatabaseTable(tableName = "users")
public class User extends DatabaseObject {

    @DatabaseField(columnName = "user_id", generatedId = true, canBeNull = false)
    private int id;

    @DatabaseField(columnName = "name", canBeNull = false)
    private String name;

    @ForeignSingletonField
    private Image image;

    @ForeignCollectionField
    private Collection<Product> products;
    
    @Override
    public String toString() {
        return "User{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", products=" + products +
                '}';
    }
}
```
```java
@AllArgsConstructor
@NoArgsConstructor
@Data
@DatabaseTable(tableName = "images")
public class Image extends DatabaseObject {

    @DatabaseField(columnName = "id", generatedId = true)
    @Id
    private int id;

    @DatabaseField(columnName = "image_url", canBeNull = false)
    private String imageUrl;

    @DatabaseField(foreignColumnName = "user_id", foreign = true, foreignAutoCreate = true, foreignAutoRefresh = true)
    private User user;
}
```

I need an annotation like @ForeignCollectionField which is for single object @ForeignSingletonField. I tried to make a one-to-one relationship with javax.persistence annotation with my knowledge from jpa, but I guess ormlite doesn't support persistence annotation enough.

If this ormlite is also not possible I'm calling out to the ORMLite developers. It would be great if you add @ForeignSingletonField annotation .

The annatation I want you to add corresponds to @OneToOne in JPA

**Comment by j256** [comment]
What would the resulting schema look like?  Also, can you replace the images with code samples?

**Reply by birolozturkk** [reply]
> What would the resulting schema look like? Also, can you replace the images with code samples?

@ForeignCollectionField annotation made for single objects

**Reply by j256** [reply]
Can you give a concrete example of the SQL?

**Reply by birolozturkk** [reply]
![image](https://user-images.githubusercontent.com/70208466/157102376-df715bae-d267-427f-911f-ab101dff6c7b.png)
![image](https://user-images.githubusercontent.com/70208466/157102396-d6f5bc5b-bce6-4979-9ecc-70d8ef49e60e.png)

**Reply by birolozturkk** [reply]
do you understand my suggestion

**Comment by birolozturkk** [comment]
Will my suggestion be considered?

**Comment by j256** [comment]
Sorry for the delay on this.  Hasn't been a priority to add it but it's on the list.
